internal/provider/resource_apply_module.go

```diff
@@ -0,0 +1,589 @@
+func (t *applyModuleResource) Configure(_ context.Context,
+	req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	t.client = req.ProviderData.(*tharsis.Client)
+}
+
+func (t *applyModuleResource) Create(ctx context.Context,
+	req resource.CreateRequest, resp *resource.CreateResponse) {
+
+	// Retrieve values from apply module.
+	var applyModule ApplyModuleModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &applyModule)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Do plan and apply, no destroy.
+	var created ApplyModuleModel
+	resp.Diagnostics.Append(t.doRun(ctx, &doRunInput{
+		model: applyModule,
+	}, &created)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Update the plan with the computed attribute values.
+	created.ID = types.StringValue(uuid.New().String())
+	resp.Diagnostics.Append(t.copyApplyModule(ctx, &created, &applyModule)...)
+
+	// Set the response state to the fully-populated plan, whether or not there is an error.
+	resp.Diagnostics.Append(resp.State.Set(ctx, applyModule)...)
+}
+
+func (t *applyModuleResource) Read(ctx context.Context,
+	req resource.ReadRequest, resp *resource.ReadResponse) {
+
+	// Get the current state.
+	var state ApplyModuleModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var applied ApplyModuleModel
+	resp.Diagnostics.Append(t.getCurrentApplied(ctx, state, &applied)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Update the state with the computed attribute values.
+	resp.Diagnostics.Append(t.copyApplyModule(ctx, &applied, &state)...)
+
+	// TODO: Eventually, when the API and SDK support speculative plan runs with a module source,
+	// this should do a speculative plan run here to determine whether changes are needed.
+
+	// Set the refreshed state, whether or not there is an error.
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (t *applyModuleResource) Update(ctx context.Context,
+	req resource.UpdateRequest, resp *resource.UpdateResponse) {
+
+	// Retrieve values from plan.
+	var plan ApplyModuleModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	// Capture the original ID in order to restore it later.
+	originalID := plan.ID.ValueString()
+
+	// TODO: Please note that when the API and SDK support speculative runs with a module source,
+	// this will need to look at the results from the Read method's speculative run to determine
+	// whether to do an update.  A way will have to be found to force Terraform to allow the update.
+
+	// Do the run.
+	var updated ApplyModuleModel
+	resp.Diagnostics.Append(t.doRun(ctx, &doRunInput{
+		model: plan,
+	}, &updated)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	// Restore the original ID.  It was empty at this point.
+	updated.ID = types.StringValue(originalID)
```

Comment:
Doesn't looks like this ID should need to be set here since it's already set on the plan.